pinot-plugins/pinot-segment-writer/pinot-segment-writer-file-based/src/main/java/org/apache/pinot/plugin/segmentwriter/filebased/FileBasedSegmentWriter.java

```diff
@@ -141,6 +141,7 @@ private void resetBuffer()
   @Override
   public void collect(GenericRow row)
       throws IOException {
+    // TODO: Revisit whether we should transform the row
```

Comment:
Can you explain what the TODO means?

Reply:
In `flush()` we will use the collected row to create a segment, which applies the transforms again